plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractTrinoCatalog.java

```diff
@@ -217,9 +229,46 @@ protected SchemaTableName createMaterializedViewStorageTable(ConnectorSession se
 
         String storageSchema = getStorageSchema(definition.getProperties()).orElse(viewName.getSchemaName());
         SchemaTableName storageTable = new SchemaTableName(storageSchema, storageTableName);
-        List<ColumnMetadata> columns = definition.getColumns().stream()
-                .map(column -> new ColumnMetadata(column.getName(), typeForMaterializedViewStorageTable(typeManager.getType(column.getType()))))
-                .collect(toImmutableList());
+
+        Schema schemaWithTimestampTzPreserved = schemaFromMetadata(mappedCopy(
+                definition.getColumns(),
+                column -> {
+                    Type type = typeManager.getType(column.getType());
+                    if (type instanceof TimestampWithTimeZoneType timestampTzType && timestampTzType.getPrecision() <= 6) {
+                        // For now preserve timestamptz columns so that we can parse partitioning
+                        type = TIMESTAMP_TZ_MICROS;
+                    }
+                    else {
+                        type = typeForMaterializedViewStorageTable(type);
```

Comment:
No need to update the TODO comment in `typeForMaterializedViewStorageTable` method? 
```
        if (type instanceof TimestampWithTimeZoneType) {
            // Iceberg does not store the time zone
            // TODO allow temporal partitioning on these columns, or MV property to
            //  drop zone info and use timestamptz directly
            return VARCHAR;
        }
```